hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1458,11 +1487,33 @@ bool MDMParser::getSignalStrength(NetStatus &status)
             _net.act = ACT_GSM;
         }
 
-        sendFormated("AT+CSQ\r\n");
-        if (RESP_OK == waitFinalResp(_cbCSQ, &_net, CSQ_TIMEOUT)) {
-            ok = true;
-            status = _net;
+        // AT command used to collect signal stregnth is different for R410M radio
+        if (_dev.dev == DEV_SARA_R410) {
+            sendFormated("AT+UCGED=5\r\n");
+            if (RESP_OK != waitFinalResp(nullptr, nullptr, UCGED_TIMEOUT)) {
+                goto cleanup;
+            }
+            // Reset the ucged parsing flags before checking for UCGED command
+            ucged_parse_rsrp_ok = false;
+            ucged_parse_rsrq_ok = false;
+            sendFormated("AT+UCGED?\r\n");
+            if (RESP_OK == waitFinalResp(_cbUCGED, &_net, UCGED_TIMEOUT)) {
+                if (ucged_parse_rsrp_ok && ucged_parse_rsrq_ok)
+                {
+                    ok = true;
+                    status = _net;
+                }
+            }
```

Comment:
`if !(ucged_parse_rsrp_ok && ucged_parse_rsrq_ok)` we should be ensuring rsrp AND rsrq are set to 255.  Again, this can all be handled in _cbUCGED.